reasoner/resolution/ResolverRegistry.java

```diff
@@ -208,18 +210,25 @@ public ResolverView registerResolvable(Resolvable<?> resolvable) {
     public Actor.Driver<BoundConcludableResolver> registerBoundConcludable(
             Concludable concludable, ConceptMap bounds, Map<Actor.Driver<ConclusionResolver>, Rule> resolverRules,
             LinkedHashMap<Actor.Driver<ConclusionResolver>, Set<Unifier>> applicableRules,
-            Actor.Driver<? extends Resolver<?>> root, int iteration) {
+            Actor.Driver<? extends Resolver<?>> root, int iteration, boolean explain) {
         LOG.debug("Register BoundConcludableResolver, pattern: {} bounds: {}", concludable.pattern(), bounds);
-        Actor.Driver<BoundConcludableResolver> resolver = Actor.driver(driver -> new BoundConcludableResolver(
-                driver, concludable, bounds, resolverRules, applicableRules, this, traversalEngine, conceptMgr, resolutionTracing
-        ), executorService);
+        Actor.Driver<BoundConcludableResolver> resolver;
+        if (explain) {
```

Comment:
I think the end state we want is a single `ConcludableResolver` class, with all its sub-actors as nested classes. At that point I think we should question constructing all actors here in the registry and instead have the Coordinator (which is probably the `ConcludableResolver` class itself) do the construction. This `explain` flag way looks weird to me, but I didn't want yet another method in the registry for the two types.

Reply:
I'm inclined to agree. The reason for the registry remains to create the shared coordinators (I think) for rule and concludable. The coordinator classes, when terminated from the registry, will then terminate their children. We'll have to flesh out the termination into a bit less of a hack into a proper set of Requests/Responses. Goes to show we'll need multiple non-answer messages: materialise and terminate (that i can see now)

Reply:
Good point, that makes sense